SpiffWorkflow/task.py

```diff
@@ -78,26 +90,26 @@ def __init__(self, workflow, task_spec, parent=None, state=TaskState.MAYBE, id=N
         self._state = state
 
         self.triggered = False
-        self.task_spec = task_spec
+        self.task_spec = task_spec if task_spec else StartTask(workflow.spec)
```

Comment:
Not sure about this, check by pyright/mypy introduces the possibility of task_spec being None, but in the case of None task_spec, a lot of stuff cannot properly work.

I can delete this "fix" for typing, but it is something to consider.